### PR TITLE
Event cancellation frontend

### LINKED ISSUE #82: Implement Event Cancellation
**Branch:** `event-cancellation-frontend`

### Task at Hand
1. View the [mid-fidelity designs](https://www.figma.com/file/58CoTkFT4TgalsOyc1Patl/%F0%9F%8C%BE-Lagos-Food-Bank?type=design&node-id=1809-21467&mode=design&t=QD3MBT6azmzjOBjZ-0) for the View Events pages.
2. Implement the `EventCancelForm` component so that it looks like the following:
Image
4. Confirm that your code and pull request follow the [Pull Request Checklist](https://www.notion.so/h4i/Pull-Request-Checklist-910d5e6925df49569067f00f6aa0c0f3).
### Acceptance Criteria
- [ ] Users can see the Event Cancel Form. This must be mobile friendly (see Figma)
- [ ] Clicking on the button opens the modal

### Notes
- Use the following components: `IconText`, `EventDetails`, `Checkbox`, `Button`, `TextField`

### PR DESCRIPTION
**Pull request by bananahz**
## Summary
- Implements `EventCancellationForm`. 
- Implements clickable button that pops up modal. 
- Closes #82.

### Screenshots
Web View: 
image

Mobile: 

## Testing
It is recommended to use a `text.tsx` as a page to view the component. For example,
```
import React from "react";
import EventCancelForm from "@/components/organisms/EventCancelForm";

const Test = () => {
  return (
    <EventCancelForm eventid="1"/>
  );
};

export default Test;
```
To visualize `EventDetails` as a grey box, insert this line in `EventDetails.tsx`
```
return <div className="h-40 bg-gray-300">event details</div>;
```

## Notes
For CancelForm, we modified `MultiTextField`. We can't modify the fontweight of label and cannot add placeholder without modifying it.

### CODE REVIEW

**Review by akinfelami — CHANGES_REQUESTED** [review]
Hi! Good work on this so far! quickly:
- Can you share how you tested this? I can see screenshots of the web site but I would want to see it myself

**Comment by Y1chenYao** [comment]
> Hi! Good work on this so far! quickly:
> 
> * Can you share how you tested this? I can see screenshots of the web site but I would want to see it myself

Hi Akin! I've edited the description of the pr. :)

File: frontend/src/components/organisms/EventCancelForm.tsx [hunk 1]
```diff
@@ -1,14 +1,120 @@
...
+        <div className="col-start-1 col-end-5 pt-4 md:col-start-2 md:col-end-4 md:pt-8">
+          <Button
+            children="Cancel Registration"
+            color="gray"
+            onClick={handleOpen}
+          ></Button>
+        </div>
+      </div>
+    </div>
+  );
```
Comment (akinfelami) [inline frontend/src/components/organisms/EventCancelForm.tsx:117]: Can you please wrap the Textfield in a `form`? You can copy from login and signup forms where I use react-hook-form. The "reason for canceling" has the asterisk to indicate that it is required but now it is not enforced. The form will allow us to achieve that!

**Review by akinfelami — CHANGES_REQUESTED** [review]
Great Job guys!

I think this looks good to me. Just one last final change. Please see the comments. I notice you actually import `useForm` hook but you never use it.

File: frontend/src/components/organisms/EventCancelForm.tsx (hunk 1 above)
Comment (bananahz) [reply frontend/src/components/organisms/EventCancelForm.tsx]: I think useForm is used, did you mean submitHandler?

**Review by louisvalencia — APPROVED** [review]
Looks good to me 👍🏽

**Review by DannyThorne01 — APPROVED** [review]

**Review by akinfelami — APPROVED** [review]
lgtm

**Review by jasozh — APPROVED** [review]
Good work! I changed the **Return to My Events** button to use `IconText` to avoid code duplication. Also, make sure you use spaces instead of tabs for indentation. My autoformatter had to reformat every line of code due to this.

**Pull request merged**